### DISCUSSION TITLE
ETCD Timestamp Format Conversion

### DISCUSSION THREAD

**Discussion by ankitg-03** [uber-go/zap#1206] [Q&A] [answered]
Hi,
Could you please help to change the timestamp format as below.
Currently we get ETCD timestamps in the the following format - 2022-11-18T07:16:40.169Z but, our required format is 2022-11-18T07:15:24.722+02:00
Is there is a way to get timestamp in ETCD log in the required format 2022-11-18T07:15:24.722+02:00 ("yyyy-MM-dd'T'HH:mm:ss.SSSxxx") through some ETCD configuration?

**Comment by ankitg-03** [comment]
Hi
I wanted to follow up on the above mentioned query.

**Comment by abhinav** [comment, marked as answer]
Hey @ankitg-03, we don't control the etcd format.
You'll want to make a request on etcd's Issues/Discussions.
They configure Zap's timestamp output in two places:

- server: https://github.com/etcd-io/etcd/blob/6f899a7b40f7631461ffeda0067aa4c42dd17812/etcdutl/etcdutl/common.go#L28
- client: https://github.com/etcd-io/etcd/blob/3c64ae4b7941c5a377e011aae51e5e29da7d0655/client/pkg/logutil/zap.go#L59-L62
